### PR TITLE
[SPARK-23394][UI] In RDD storage page show the executor addresses instead of the IDs

### PR DESCRIPTION
**Pull request by attilapiros**
## What changes were proposed in this pull request?

Extending RDD storage page to show executor addresses in the block table. 

## How was this patch tested?

Manually:

screen shot 2018-02-13 at 10 30 59

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala [hunk 1]
```diff
@@ -198,7 +200,7 @@ private[ui] class BlockDataSource(
       rddPartition.storageLevel,
       rddPartition.memoryUsed,
       rddPartition.diskUsed,
-      rddPartition.executors.mkString(" "))
+      rddPartition.executors.map(id => executorIdToAddress.get(id).getOrElse(id)).mkString(" "))
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: `.map { id => ... }`

Also in other places.

File: core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala [hunk 2]
```diff
@@ -243,7 +246,8 @@ private[ui] class BlockPagedTable(
     rddPartitions,
     pageSize,
     sortColumn,
-    desc)
+    desc,
+    executorSummaries.map(ex => (ex.id, ex.hostPort)).toMap)
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: Probably a good idea to keep the map sorted by the executor id, somehow.

**Comment by vanzin** [comment]
ok to test

**Comment by kiszk** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala [hunk 3]
```diff
@@ -76,7 +76,8 @@ private[ui] class RDDPage(parent: SparkUITab, store: AppStatusStore) extends Web
         rddStorageInfo.partitions.get,
         blockPageSize,
         blockSortColumn,
-        blockSortDesc)
+        blockSortDesc,
+        store.executorList(false))
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: Should this only list active executors?

Theoretically you shouldn't have any blocks stored on dead executors.

Comment (attilapiros) [reply core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: As we have a fallback to executorId we can do that.

**Comment by attilapiros** [comment]
retest this please

**Comment by kiszk** [comment]
retest this please

**Comment by attilapiros** [comment]
jenkins retest this please

**Comment by attilapiros** [comment]
jenkins retest this please

File: core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala [hunk 4]
```diff
@@ -198,7 +200,10 @@ private[ui] class BlockDataSource(
       rddPartition.storageLevel,
       rddPartition.memoryUsed,
       rddPartition.diskUsed,
-      rddPartition.executors.mkString(" "))
+      rddPartition.executors
+        .sorted
+        .map { id => executorIdToAddress.get(id).getOrElse(id) }
```
Comment (jiangxb1987) [inline core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: nit: should we sort by address?

**Comment by jiangxb1987** [comment]
LGTM

File: core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala (hunk 4 above)
Comment (attilapiros) [reply core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: ok ,done

**Comment by vanzin** [comment]
Merging to master / 2.3.

**Pull request closed without merging**